### ISSUE TITLE
Support for eager loading

### ISSUE THREAD

**Issue by cec** [derekharmel/sunspot_mongo#8]
Hi,

I noticed that eager loading does not work.

I have this model:

```
class Poi
  include Mongoid::Document
  has_and_belongs_to_many :categories
  ....
end
```

running 

```
Poi.solr_search(include: :categories)
```

raises the following error:

```
NoMethodError: undefined method `include=' for #<Sunspot::Mongo::DataAccessor:0x007f8caac11c20>
```

Is eagerloading supported by sunspot-mongo? 

Thanks!

P.S.: I know that many-to-many is not well-viewed in mongo, but I need it here...

**Comment by dblock** [comment]
It _should_ work, aka if it doesn't it's a bug.

**Comment by cec** [comment]
Then it's a bug.

Will anybody have time to fix this in the near future?
Since mongo does not allow to return documents in the order of the ids that were specified, preserving the score order would be require iterating on the results to map them to the ids returned by sunspot.
It feels so depressing to do in this day and age.

Is there any information I can provide to help besides the following?

Ruby 2.2
mongoid 4.0.2
sunspot_mongo 1.1.0

**Comment by dblock** [comment]
I don't think anyone here will jump on this soon but I am happy to help. Start by writing a test that helps reproduce the problem?

**Comment by cec** [comment]
Hi @dblock, 

I had a look at how sunspot_rails does this for ActiveRecord; it turns out it's very simple.

I hope to make a PR by tomorrow morning.

**Comment by cec** [comment]
Hey @dblock,

in the end yesterday I had more pressing matters, but  now I created my PR #9 

Let me know what you think about it

**Comment by dblock** [comment]
Fixed in #9.
